src/base/memory_monitor_sr.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_monitor_sr.cpp - Implementation of memory monitor module
+ */
+
+#include <cstring>
+#include <algorithm>
+
+#include "openssl/md5.h"
+#include "memory_monitor_sr.hpp"
+
+namespace cubmem
+{
+  memory_monitor::memory_monitor (const char *server_name)
+    : m_server_name {server_name} {}
+
+  std::string memory_monitor::make_tag_name (const char *file, const int line)
+  {
+    std::string filecopy (file);
+    std::string ret;
+
+    // Find the last occurrence of "src" in the path
+    size_t pos = filecopy.rfind ("src");
```

Comment:
What if the name of a directory of file includes `src`? If it's not the case for now, it make "src" as a special string that we can't use for files.

I know it's a kind of "requirement", but do we really need to includes all the path starting with the `src`? I think the file name is good enough. Please share your opinion on this.

Reply:
"rfind(str)" method find the last matching str that the function meets. So I think this cannot be a matter.

Reply:
What if 
- `src/a/module_a_src.c`
-  `src/a/a_src_b/.c`
-  `src/a/a_descriptor.c`,, oh, we have `record_descriptor.cpp`. Does it work with this?

Reply:
After considering it, I have decided to maintain the existing method. However, to address the scenario you mentioned in your comment, I changed "/src/" to the target string of rfind. Additionally, in the path of the 3rd-party code, "/src/" may exist. To internally verify this, when building in debug mode, I compare the positions of the leftmost and rightmost occurrences of "/src/" to ensure that only one "/src/" exists. This decision was made after discussion with @hornetmj.

Reply:
Honestly, I'm still against this in that I doubt this job is worth it with the cost by many string copies and comparisons. This is a function called every time a memory is allocated, isn't it?

Anyway, that's just my concern and I'll follow your decision. We can revisit if it really matters :).

Reply:
> with the cost by many string copies and comparisons.

I aggree with this comment. One of option I think is to use c style approach using char * and strrstr(). Let's see the result of performance test and visit again if it is necessary.